timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -767,6 +767,9 @@ def leaderboard(self, data: Optional[TimeSeriesDataFrame] = None) -> pd.DataFram
                     model_info[model_name]["score_test"] = float("nan")
                     model_info[model_name]["pred_time_test"] = float("nan")
 
+        if len(model_info) == 0:
```

Comment:
Let's imagine a scenario where someone is looping through different predictors fit to the same data, and collecting their leaderboards. When asking for this job's leaderboard, why should an exception (to say, something went wrong in calculating the leaderboard -- and especially a ValueError to say something went wrong because the customer provided a wrong input argument to this function) be thrown? Isn't an empty leaderboard the semantically correct response?

I would rather just raise a UserWarning and return the empty dataframe.

Reply:
I agree, this is a much more elegant solution. Currently, if no models were trained, user receives following output:
```
Traceback (most recent call last):
  File "/local/home/shchuro/workspace/autogluon/shchuro/train.py", line 43, in <module>
    predictor.leaderboard(d_test)
  File "/local/home/shchuro/workspace/autogluon/timeseries/src/autogluon/timeseries/predictor.py", line 621, in leaderboard
    leaderboard = self._learner.leaderboard(data)
  File "/local/home/shchuro/workspace/autogluon/timeseries/src/autogluon/timeseries/learner.py", line 367, in leaderboard
    return self.load_trainer().leaderboard(data)
  File "/local/home/shchuro/workspace/autogluon/timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py", line 773, in leaderboard
    df.sort_values(by=[sort_column, "model"], ascending=[False, False], inplace=True)
  File "/home/shchuro/miniconda3/envs/ag/lib/python3.9/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/shchuro/miniconda3/envs/ag/lib/python3.9/site-packages/pandas/core/frame.py", line 6304, in sort_values
    keys = [self._get_label_or_level_values(x, axis=axis) for x in by]
  File "/home/shchuro/miniconda3/envs/ag/lib/python3.9/site-packages/pandas/core/frame.py", line 6304, in <listcomp>
    keys = [self._get_label_or_level_values(x, axis=axis) for x in by]
  File "/home/shchuro/miniconda3/envs/ag/lib/python3.9/site-packages/pandas/core/generic.py", line 1840, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'score_val'
```
I will update the PR to follow your suggestion!